source/_ext/build_engine_schema_docs.py

```diff
@@ -0,0 +1,193 @@
+import os
+from typing import Any, Dict
+
+from howso.client.api import get_api
+from jinja2 import BaseLoader, Environment
+from jinja2.exceptions import UndefinedError
+from sphinx.util import logging
+
+logger = logging.getLogger(__name__)
+
+
+def setup(app):
+    """
+    Setup function for the Sphinx extension.
+    This builds RST pages for the needed Engine type schemas.
+    """
+    def generate_rst_from_json(app):
+        """
+        Function that runs before Sphinx starts reading source files.
+        Place your JSON fetching and RST generation code here.
+        """
+        try:
+            needed_schemas = [
+                "FeatureAttributes",
+                "FeatureBounds",
+                "FeatureDataType",
+                "FeatureOriginalType",
+                "FeatureOriginalTypeBoolean",
+                "FeatureOriginalTypeDate",
+                "FeatureOriginalTypeDatetime",
+                "FeatureOriginalTypeInteger",
+                "FeatureOriginalTypeNumeric",
+                "FeatureOriginalTypeObject",
+                "FeatureOriginalTypeString",
+                "FeatureOriginalTypeTime",
+                "FeatureOriginalTypeTimedelta",
+                "FeatureType",
+            ]
+
+            # THIS WILL NOT WORK IF HOWSO-ENGINE-PY IS INSTALLED EDITABLY
+            # (unless you have the .caml in your local repo too)
+            schema_map = get_api()['schemas']
+            os.makedirs("howso/types", exist_ok=True)
+            for schema_name in needed_schemas:
+                schema = schema_map[schema_name]
+                if schema is None:
+                    print(schema_name)
+                    continue
+                try:
+                    rst_output = generate_sphinx_doc(schema, schema_name)
+                except UndefinedError as e:
+                    print(schema_name)
+                    print(schema)
+                    print(e)
+                    continue
+                with open(f'howso/types/{schema_name}.rst', 'w') as f:
+                    f.write(rst_output)
+
+            logger.info('Successfully generated RST files from JSON data')
+        except Exception as e:
+            logger.error(f'Error generating RST files: {str(e)}')
+            raise
+
+    # Connect to the builder-inited event
+    app.connect('builder-inited', generate_rst_from_json)
+
+    return {
+        'version': '0.1',
+        'parallel_read_safe': True,
+        'parallel_write_safe': True,
+    }
+
+
+# Define our base template as a string
+BASE_TEMPLATE = """
+{{ title }}
+{{ '=' * title|length }}
+
+.. contents:: Table of Contents
+   :depth: 3
+
+{% if 'description' in json_spec %}
+Overview
+--------
+{{ json_spec.description }}
+{% endif %}
+
+{% if 'type' in json_spec %}
+:type: {{ 'object' if json_spec.type == 'assoc' else json_spec.type }}
+{% endif %}
+
+
+{% if 'any_of' in json_spec %}
+:One of:
```

Comment:
```suggestion
:One of:

```
Need a new line after this to fix a warning in the build output